update README files for security components (#1978)
edgexfoundry/edgex-go: SECURITY.md

# Security Components in EdgeX Foundry Services

> Starting from Fuji release, EdgeX includes enhanced security features. There are 3 major components that are responsible for security

Author: fixed.

features: 

- [Security-secrets-setup] (https://github.com/edgexfoundry/edgex-go/blob/master/cmd/security-secrets-setup/README.md)
- [Security-secretstore-setup] (https://github.com/edgexfoundry/edgex-go/blob/master/cmd/security-secretstore-setup/README.md)

> - [Security-proxy-setup] (https://github.com/edgexfoundry/edgex-go/blob/master/cmd/security-proxy-setup/README.md)

Reviewer: The space between the right bracket and left paren is causing this to render incorrectly.  Also, would a relative URL to README.md work?

Author: Fixed.

> Security-secrets-setup is responsible for creating necessary certificates. Security-secretstore-setup is responsible for initializing the secret store that holds various credentials for EdgeX. Security-proxy-setup is responsible for initializating the environment for proxy for EdgeX to protect all the REST API resources, set up related permissions and authentication methods. 

Author: Fixed.

# Get Started

To get started, fetch the latest docker-compose.yml and start the EdgeX containers:

```sh
wget https://raw.githubusercontent.com/edgexfoundry/developer-scripts/master/releases/fuji/compose-files/docker-compose-fuji-1.1.0.yml
docker-compose up -d
```

> Once the EdgeX is up and running, the following steps are required to access the resources of EdgeX.

Author: fixed.

1. The user needs to create an access token and associate every REST request with the security token

> while sending the request. Use "admin" as groupname below as it is the default enable group in the configuration of proxy.

Author: fixed.

```sh
docker-compose run security-proxy-setup --useradd=<account> --group=<groupname>
```

> which will create an access token. The token will be like this: eyJpc3MiOiI5M3V3cmZBc0xzS2Qwd1JnckVFdlRzQloxSmtYOTRRciIsImFjY291bnQiOiJhZG1pbmlzdHJhdG9yIn0

Reviewer 2: ```suggestion
which will create an access token. One example of an access token will be like: eyJpc3MiOiI5M3V3cmZBc0xzS2Qwd1JnckVFdlRzQloxSmtYOTRRciIsImFjY291bnQiOiJhZG1pbmlzdHJhdG9yIn0
```

Author: fixed.

> 2. The exported external ports (such as 48080, 48081 etc.) will be inaccessible due to security enhancement. Instead all the REST requests need to go through the proxy, and the proxy will redirect the request to individual microservice on behalf of the user.

Reviewer 3: s/due to/due to the
s/microservice/microservices

Author: fixed.

> E.g, if we need to send a request to the ping endpoint of coredata, the original request be like this below:

Reviewer 3: s/be like this below/was as below:

Reviewer 4: Consider s/the original request be like this below/such as the following

Author: fixed.

```
curl http://{coredata-microservice-ip}:48080/api/v1/ping
```

> Such request needs to be converted into a request like this below:

Reviewer 3: s/Such request needs to be converted into a request like this below:/Such a request is converted as below:

> Such request needs to be converted into a request like this below:

Reviewer 2: ```suggestion
Such request now needs to be converted into a request through `proxy service` and provided `access token`.  One example is like this below:
```

Author: fixed.

```

> curl -k https://{proxy-service-ip}:8443/coredata/api/v1/ping -H "Authorization: Bearer <access-token>"

Reviewer 3: ```
Please note in the above the use of the proxy IP address,  its port and the need to specify the authorisation header and  the access token therein.

Author: fixed.
